src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1331,6 +1331,177 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
+                Entity entity = new(Source: new("get_books", EntitySourceType.StoredProcedure, null, null),
+                              Rest: new(new SupportedHttpVerb[] { SupportedHttpVerb.Get, SupportedHttpVerb.Post }),
+                              GraphQL: new(Singular: "GetBooks", Plural: "GetBooks"),
+                              Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                              Relationships: null,
+                              Mappings: null
+                             );
+
+                string entityName = "GetBooks";
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName);
+            }
+            else
+            {
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions);
+            }
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+
+            AuthenticationOptions AuthenticationOptions = new(Provider: EasyAuthType.StaticWebApps.ToString(), null);
+            HostOptions staticWebAppsHostOptions = new(null, AuthenticationOptions);
+
+            RuntimeOptions runtimeOptions = configuration.Runtime;
+            RuntimeOptions baseRouteEnabledRuntimeOptions = new(runtimeOptions.Rest, runtimeOptions.GraphQL, staticWebAppsHostOptions, "/data-api");
+            RuntimeConfig baseRouteEnabledConfig = configuration with { Runtime = baseRouteEnabledRuntimeOptions };
+            File.WriteAllText(CUSTOM_CONFIG, baseRouteEnabledConfig.ToJson());
+
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(SupportedHttpVerb.Post);
+                HttpRequestMessage request = new(httpMethod, requestPath);
+                if (entityType is not EntitySourceType.StoredProcedure)
+                {
+                    string requestBody = @"{
+                        ""title"": ""Harry Potter and the Order of Phoenix"",
+                        ""publisher_id"": 1234
+                    }";
+
+                    JsonElement requestBodyElement = JsonDocument.Parse(requestBody).RootElement.Clone();
+                    request = new(httpMethod, requestPath)
+                    {
+                        Content = JsonContent.Create(requestBodyElement)
+                    };
+                }
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
+
+                string locationHeader = response.Headers.Location.AbsoluteUri;
+                Assert.IsTrue(locationHeader.StartsWith(expectedLocationHeader));
```

Comment:
Why check for StartsWith as opposed to Equals?

Reply:
When `base-route` is configured, the Location header in the repsonse will be of the form `/<base-route>/api/EntityName/id/<PKs>`. However, `base-route` is applicable only in SWA-DAB integrated scenario. When DAB engine is run independently, all the APIs are hosted on the `/api` endpoint. So, we cannot directly perform a GET request on the Location header received in the response (like we do in the test where `base-route` is absent).

To test whether the returned `base-route` returned in the response is correct, I am asserting it against the `expectedLocationHeader`. Since, the PK for the `Book` entity is auto-generated, it is not possible to know the exact `id` beforehand. So,  `expectedLocationHeader` string does not have the exact id (is of the form `.../id/`). `Equals` would not work here and hence decided to go with `startsWith`. Also, to validate that the id returned in the response is valid, another GET request is performed after removing the base-route from the URL and checking that it results in an OK response (200).

Reply:
is there a structure like AbsoluteUri that can be contructed with the value in your location header? then you can more efficiently(not writing a custom algo) check any remaining components of the uri.

Reply:
It is possible to get the individual components of the URI (we get the complete URI from the response header) by assigning it to URI class and writing a little bit of custom logic to identify the base-route, Entity name etc. But, introducing a custom logic cannot be avoided.

My thought process behind testing the location header this way are
a) The intention of returning a Location header is to facilitate performing a GET request to fetch the details of the newly created item. So, I had added a validation that performs a GET request using the location header.
b) Since the test runs DAB independently (not integrated with SWA), a separate validation is needed for base-route (a direct GET request cannot be performed on the location header). Base route is validated by asserting against expected values. 

IMHO, this does not introduce a long running test (just one additional GET request) - which should be fine given the nature of the field being validated.